### ISSUE TITLE
Does ai.djl.training.loss.SoftmaxCrossEntropyLoss work correctly?

### ISSUE THREAD

**Issue by SuperMaskv** [deepjavalibrary/djl#2492]
I'm using DJL v0.21.0
My prediction is in shape of [1, 462, 4], and my label is in the shape of [1, 462]. And SoftmaxCrossEntropyLoss works with default init parameters.
![image](https://user-images.githubusercontent.com/25413481/228826485-138cab52-621a-43f9-9935-8893b06519e0.png)

I've tried to create a custom CELoss like below:
```java
    @Override
    public NDArray evaluate(NDList labels, NDList predictions) {
        int classAxis = -1;
        NDArray pred = predictions.singletonOrThrow().logSoftmax(-1).argMax(-1).toType(DataType.FLOAT32, false);
        NDArray lab = labels.singletonOrThrow().reshape(pred.getShape());
        NDArray loss = pred.mul(lab).neg().sum(new int[] {classAxis}, true);
        return loss.mean();
    }
```
but I got another exception:
```bash
ai.djl.engine.EngineException: element 0 of tensors does not require grad and does not have a grad_fn
	at ai.djl.pytorch.jni.PyTorchLibrary.torchBackward(Native Method) ~[pytorch-engine-0.21.0.jar:na]
	at ai.djl.pytorch.jni.JniUtils.backward(JniUtils.java:1634) ~[pytorch-engine-0.21.0.jar:na]
	at ai.djl.pytorch.engine.PtGradientCollector.backward(PtGradientCollector.java:70) ~[pytorch-engine-0.21.0.jar:na]
	at ai.djl.pytorch.engine.PtGradientCollector.backward(PtGradientCollector.java:55) ~[pytorch-engine-0.21.0.jar:na]
	at ai.djl.training.EasyTrain.trainSplit(EasyTrain.java:125) ~[api-0.21.0.jar:na]
	at ai.djl.training.EasyTrain.trainBatch(EasyTrain.java:110) ~[api-0.21.0.jar:na]
	at ai.djl.training.EasyTrain.fit(EasyTrain.java:58) ~[api-0.21.0.jar:na]
```

**Comment by SuperMaskv** [comment]
I've tried to create a custom CELoss like below:
```java
    @Override
    public NDArray evaluate(NDList labels, NDList predictions) {
        int classAxis = -1;
        NDArray pred = predictions.singletonOrThrow().logSoftmax(-1).argMax(-1).toType(DataType.FLOAT32, false);
        NDArray lab = labels.singletonOrThrow().reshape(pred.getShape());
        NDArray loss = pred.mul(lab).neg().sum(new int[] {classAxis}, true);
        return loss.mean();
    }
```
but I got another exception:
```bash
ai.djl.engine.EngineException: element 0 of tensors does not require grad and does not have a grad_fn
	at ai.djl.pytorch.jni.PyTorchLibrary.torchBackward(Native Method) ~[pytorch-engine-0.21.0.jar:na]
	at ai.djl.pytorch.jni.JniUtils.backward(JniUtils.java:1634) ~[pytorch-engine-0.21.0.jar:na]
	at ai.djl.pytorch.engine.PtGradientCollector.backward(PtGradientCollector.java:70) ~[pytorch-engine-0.21.0.jar:na]
	at ai.djl.pytorch.engine.PtGradientCollector.backward(PtGradientCollector.java:55) ~[pytorch-engine-0.21.0.jar:na]
	at ai.djl.training.EasyTrain.trainSplit(EasyTrain.java:125) ~[api-0.21.0.jar:na]
	at ai.djl.training.EasyTrain.trainBatch(EasyTrain.java:110) ~[api-0.21.0.jar:na]
	at ai.djl.training.EasyTrain.fit(EasyTrain.java:58) ~[api-0.21.0.jar:na]
```
